### DISCUSSION TITLE
ProcessPending set different destination per file

### DISCUSSION THREAD

**Discussion by james-hinton** [rpldy/react-uploady#410] [Q&A]
**Describe the bug**
I want to be able to set a different destination for each file. I have to add filename to the destination URL (Azure blob).

**Expected behavior**
That it sends the file individually, but it processes every file with the same destination URL and ends up overwriting all the files with the same file

**Versions**
1.1.0

**Code**
Sorry I really struggle to format this, it just doesnt want to format
```
      for (let i = 0; i < toDownload.length; i++) {
        const destination = {
          url:
            "https://#.blob.core.windows.net/files/" +
            toDownload[i].path +
            "?sv=2021-06-08&ss=b&srt=sco&sp=rwdlactfx&se=2022-10-21T01:04:32Z&st=2022-09-20T17:04:32Z&spr=https&sig=#",
        };

        processPending({
          destination,
          fileFilter: (file) => file.item === toDownload[i].file.item,
          headers: {
            "x-ms-blob-type": "BlockBlob",
          },
          grouped: false,
        });
      }
```

**Comment by yoavniran** [comment]
hey @james-hinton 
processPending isnt meant for individual handling of options.

What you're looking for is: [useRequestPreSend](https://react-uploady.org/docs/api/hooks/useRequestPreSend/)
it allows overriding options per file just before its uploaded.

This [guide](https://react-uploady.org/docs/guides/DynamicParameters/) should help get you started with that.

Let me know if you have questions. In the meantime ill convert this into a discussion.

**Reply by james-hinton** [reply]
Thanks for pointing me in the right direction, by changing the batch size to one I was then able to set the unique destination per file using that hook. 

```
  useRequestPreSend(async ({ items, options }) => {
    const filename = items[0].file.name;

    const sasToken = await axios.get(
      `${process.env.REACT_APP_PORTAL_BACKEND_URL}/file/sas_token/${filename}/`
    );

    return Promise.resolve({
      options: {
        destination: {
          url: sasToken.data.endpoint,
        },
        method: "PUT",
      },
    });
  });
```

**Reply by woodybriggs** [reply]
How to do this while using a batch size greater than 1? To me it would make more sense to return an array of options

**Reply by yoavniran** [reply]
@woodybriggs 

batching means sending multiple files in one request. 
Only one set of options can be used per request.
So I dont see what would an array of options do.

If you want to send information about the files, you can add an array param that will be turned into a comma delimited string that you can parse on your server:
```

options: {
   ids: [1,2,3]
}

```
